In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

# Initial EDA

In [2]:
# Load the data
df = pd.read_csv('E:\Documents\Google Drive\Eskwelabs\Sprint 5 - Capstone\data\DOE- Data Request- Sample Daily Data_GEN_2023924.csv')

C:\Users\Gaea\AppData\Local\Temp\ipykernel_21488\569495927.py:2: DtypeWarning: Columns (9,14,66,69) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('E:\Documents\Google Drive\Eskwelabs\Sprint 5 - Capstone\data\DOE- Data Request- Sample Daily Data_GEN_2023924.csv')


In [3]:
pd.set_option('display.max_columns', None)
df.head()

time_interval    delivery_date  delivery_hour    resource_id  \
0  24/09/2023 0:05  24/09/2023 0:00              1  01BTSOLEN_BAT   
1  24/09/2023 0:05  24/09/2023 0:00              1    01CNCEP_BAT   
2  24/09/2023 0:05  24/09/2023 0:00              1    01LAMAO_BAT   
3  24/09/2023 0:05  24/09/2023 0:00              1    01LIMAY_BAT   
4  24/09/2023 0:05  24/09/2023 0:00              1  01MAGAPIT_BAT   

        resource_name trading_id major_part_id_new type_id bid_type nsch_type  \
0      BATAAN BATTERY       BSEI                AC     GEN      REG       NaN   
1  CONCEPCION BATTERY       UPSI               SMC     GEN      REG       NaN   
2       LAMAO BATTERY       UPSI               SMC     GEN      REG       NaN   
3       LIMAY BATTERY       UPSI               SMC     GEN      REG       NaN   
4     MAGAPIT BATTERY       UPSI               SMC     GEN      REG       NaN   

  plant_type region_id zone_id  max_cap_rtd Outage_Tag  outage_capacity_rtd  \
0        BAT      CLUZ  NLUZON          0.5        NaN                  NaN   
1        BAT      CLUZ  NLUZON         60.0        NaN                  NaN   
2        BAT      CLUZ  NLUZON         50.0        NaN                  NaN   
3        BAT      CLUZ  NLUZON         40.0        NaN                  NaN   
4        BAT      CLUZ  NLUZON         40.0        NaN                  NaN   

   rtd_price1  rtd_quantity1  rtd_price2  rtd_quantity2  rtd_price3  \
0         NaN            NaN         NaN            NaN         NaN   
1         NaN            NaN         NaN            NaN         NaN   
2    -10000.0          -50.0    -10000.0            0.0     32000.0   
3    -10000.0          -40.0    -10000.0            0.0     32000.0   
4         NaN            NaN         NaN            NaN         NaN   

   rtd_quantity3  rtd_price4  rtd_quantity4  rtd_price5  rtd_quantity5  \
0            NaN         NaN            NaN         NaN            NaN   
1            NaN         NaN            NaN         NaN            NaN   
2           50.0         NaN            NaN         NaN            NaN   
3           40.0         NaN            NaN         NaN            NaN   
4            NaN         NaN            NaN         NaN            NaN   

   rtd_price6  rtd_quantity6  rtd_price7  rtd_quantity7  rtd_price8  \
0         NaN            NaN         NaN            NaN         NaN   
1         NaN            NaN         NaN            NaN         NaN   
2         NaN            NaN         NaN            NaN         NaN   
3         NaN            NaN         NaN            NaN         NaN   
4         NaN            NaN         NaN            NaN         NaN   

   rtd_quantity8  rtd_price9  rtd_quantity9  rtd_price10  rtd_quantity10  \
0            NaN         NaN            NaN          NaN             NaN   
1            NaN         NaN            NaN          NaN             NaN   
2            NaN         NaN            NaN          NaN             NaN   
3            NaN         NaN            NaN          NaN             NaN   
4            NaN         NaN            NaN          NaN             NaN   

   rtd_price11  rtd_quantity11  rtd_rr_break_quantity1  rtd_rr_up1  \
0          NaN             NaN                     NaN         NaN   
1          NaN             NaN                     NaN         NaN   
2          NaN             NaN                    50.0      2300.4   
3          NaN             NaN                    40.0      2642.4   
4          NaN             NaN                     NaN         NaN   

   rtd_rr_down1  rtd_rr_break_quantity2  rtd_rr_up2  rtd_rr_down2  \
0           NaN                     NaN         NaN           NaN   
1           NaN                     NaN         NaN           NaN   
2        2300.4                     NaN         NaN           NaN   
3        2642.4                     NaN         NaN           NaN   
4           NaN                     NaN         NaN           NaN   

   rtd_rr_break_quantity3  rtd_rr_up

In [4]:
df.tail()

time_interval    delivery_date  delivery_hour   resource_id  \
120667  24/09/2023 23:50  24/09/2023 0:00             24  08SLWIND_G01   
120668  24/09/2023 23:55  24/09/2023 0:00             24   08PWIND_G01   
120669  24/09/2023 23:55  24/09/2023 0:00             24  08SLWIND_G01   
120670   25/09/2023 0:00  24/09/2023 0:00             24   08PWIND_G01   
120671   25/09/2023 0:00  24/09/2023 0:00             24  08SLWIND_G01   

           resource_name trading_id major_part_id_new type_id bid_type  \
120667  SAN LORENZO WIND      TAREC                AC     GEN     NSCH   
120668   PWEI NABAS WIND       PWEI             BWCUA     GEN     NSCH   
120669  SAN LORENZO WIND      TAREC                AC     GEN     NSCH   
120670   PWEI NABAS WIND       PWEI             BWCUA     GEN     NSCH   
120671  SAN LORENZO WIND      TAREC                AC     GEN     NSCH   

       nsch_type plant_type region_id zone_id  max_cap_rtd Outage_Tag  \
120667      MUST       WIND      CVIS   PANAY         54.0        NaN   
120668      MUST       WIND      CVIS   PANAY         36.7        NaN   
120669      MUST       WIND      CVIS   PANAY         54.0        NaN   
120670      MUST       WIND      CVIS   PANAY         36.7        NaN   
120671      MUST       WIND      CVIS   PANAY         54.0        NaN   

        outage_capacity_rtd  rtd_price1  rtd_quantity1  rtd_price2  \
120667                  NaN         NaN            NaN         NaN   
120668                  NaN         NaN            NaN         NaN   
120669                  NaN         NaN            NaN         NaN   
120670                  NaN         NaN            NaN         NaN   
120671                  NaN         NaN            NaN         NaN   

        rtd_quantity2  rtd_price3  rtd_quantity3  rtd_price4  rtd_quantity4  \
120667            NaN         NaN            NaN         NaN            NaN   
120668            NaN         NaN            NaN         NaN            NaN   
120669            NaN         NaN            NaN         NaN            NaN   
120670            NaN         NaN            NaN         NaN            NaN   
120671            NaN         NaN            NaN         NaN            NaN   

        rtd_price5  rtd_quantity5  rtd_price6  rtd_quantity6  rtd_price7  \
120667         NaN            NaN         NaN            NaN         NaN   
120668         NaN            NaN         NaN            NaN         NaN   
120669         NaN            NaN         NaN            NaN         NaN   
120670         NaN            NaN         NaN            NaN         NaN   
120671         NaN            NaN         NaN            NaN         NaN   

        rtd_quantity7  rtd_price8  rtd_quantity8  rtd_price9  rtd_quantity9  \
120667            NaN         NaN            NaN         NaN            NaN   
120668            NaN         NaN            NaN         NaN            NaN   
120669            NaN         NaN            NaN         NaN            NaN   
120670            NaN         NaN            NaN         NaN            NaN   
120671            NaN         NaN            NaN         NaN            NaN   

        rtd_price10  rtd_quantity10  rtd_price11  rtd_quantity11  \
120667          NaN             NaN          NaN             NaN   
120668          NaN             NaN          NaN             NaN   
120669          NaN             NaN          NaN             NaN   
120670          NaN             NaN          NaN             NaN   
120671          NaN             NaN          NaN             NaN   

        rtd_rr_break_quantity1  rtd_rr_up1  rtd_rr_down1  \
120667                     NaN         NaN           NaN   
120668                     NaN         NaN           NaN   
120669                     NaN         NaN           NaN   
120670                     NaN         NaN           NaN   
120671                     NaN         NaN           NaN   

        rtd_rr_break_quantity2  rtd_rr_up2  rtd_rr_down2  \
120667                  

In [5]:
df_shape = df.shape
df_shape

(120672, 108)

Columns 9, 14, 66, and 69 have mixed dtypes
9    nsch_type                   object
14   Outage_Tag                  object
66   com_status                  object
69   marginal_plant              object


In [6]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120672 entries, 0 to 120671
Data columns (total 108 columns):
 #    Column                      Dtype  
---   ------                      -----  
 0    time_interval               object 
 1    delivery_date               object 
 2    delivery_hour               int64  
 3    resource_id                 object 
 4    resource_name               object 
 5    trading_id                  object 
 6    major_part_id_new           object 
 7    type_id                     object 
 8    bid_type                    object 
 9    nsch_type                   object 
 10   plant_type                  object 
 11   region_id                   object 
 12   zone_id                     object 
 13   max_cap_rtd                 float64
 14   Outage_Tag                  object 
 15   outage_capacity_rtd         float64
 16   rtd_price1                  float64
 17   rtd_quantity1               float64
 18   rtd_price2                  float64
 19   

## Fix 2 datatypes in 1 column error

In [7]:
#determine why they contain unique values
unique_9 = pd.unique(df['nsch_type'])
unique_9

array([nan, 'NSCH', 'PRI', 'MUST'], dtype=object)

In [8]:
unique_14 = pd.unique(df['Outage_Tag'])
unique_14

array([nan, 'OUT'], dtype=object)

In [9]:
unique_66 = pd.unique(df['com_status'])
unique_66

array(['NOT', nan], dtype=object)

In [10]:
unique_69 = pd.unique(df['marginal_plant'])
unique_69

array([nan, 'Y'], dtype=object)

In [11]:
#transform nan datatype into empty string
df['nsch_type'] = df['nsch_type'].fillna('').astype(str)
df['Outage_Tag'] = df['Outage_Tag'].fillna('').astype(str)
df['com_status'] = df['com_status'].fillna('').astype(str)
df['marginal_plant'] = df['marginal_plant'].fillna('').astype(str)

## Check outliers

In [12]:
#check for outliers in the data
df.describe()

delivery_hour    max_cap_rtd  outage_capacity_rtd    rtd_price1  \
count  120672.000000  120672.000000          7582.000000  62197.000000   
mean       12.500000      63.253461           132.336956  12154.107862   
std         6.922215     106.245409           146.474776  18308.400245   
min         1.000000       0.300000             2.000000 -10000.000000   
25%         6.750000       7.800000            30.000000 -10000.000000   
50%        12.500000      20.000000           116.000000  12000.000000   
75%        18.250000      70.000000           151.400000  32000.000000   
max        24.000000     668.000000           668.000000  32000.000000   

       rtd_quantity1    rtd_price2  rtd_quantity2    rtd_price3  \
count   62197.000000  62197.000000   62197.000000  29164.000000   
mean       -0.783896  12154.107862      42.750840   9117.087533   
std         5.439502  18308.400245      64.896571  13478.687784   
min       -50.000000 -10000.000000       0.000000  -9999.000000   
25%         0.000000 -10000.000000       6.500000      0.000000   
50%         0.000000  12000.000000      15.100000   3330.000000   
75%         0.000000  32000.000000      50.000000  24700.000000   
max         0.000000  32000.000000     575.000000  32000.000000   

       rtd_quantity3    rtd_price4  rtd_quantity4    rtd_price5  \
count   29164.000000  15492.000000   15492.000000   6941.000000   
mean       99.871571  13425.123170     111.314246   6052.017405   
std       117.942569  12904.295519      93.335277   7434.987626   
min         2.000000  -9998.000000       3.000000  -9997.000000   
25%        26.500000   4150.080000      55.000000   3070.000000   
50%        50.000000   6000.090000      83.700000   4512.400000   
75%       138.000000  25000.000000     120.000000   5896.000000   
max       668.000000  32000.000000     408.600000  32000.000000   

       rtd_quantity5    rtd_price6  rtd_quantity6    rtd_price7  \
count    6941.000000   4259.000000    4259.000000   3556.000000   
mean      188.150944   7318.447222     200.311458   7335.534443   
std       121.084155   8172.646403     135.459002   7463.564602   
min         4.000000   2177.000000       6.000000   2180.000000   
25%       100.000000   3183.000000     110.000000   5000.000000   
50%       150.000000   5040.900000     120.000000   5759.300000   
75%       316.000000   6500.080000     280.000000   6904.000000   
max       400.000000  32000.000000     450.000000  31999.000000   

       rtd_quantity7    rtd_price8  rtd_quantity8   rtd_price9  rtd_quantity9  \
count    3556.000000   3460.000000    3460.000000  2138.000000    2138.000000   
mean      235.810461   8349.961237     260.097341  5532.131469     303.779701   
std       152.499839   8163.005846     167.964665  1928.269498     201.158616   
min         7.000000   2204.000000       8.000000  2238.000000      10.000000   
25%       120.000000   5512.600000     130.000000  3155.000000     140.000000   
50%       135.000000   6000.000000     151.300000  5900.000000     315.000000   
75%       300.000000   7400.080000     330.000000  7800.080000     600.000000   
max       500.000000  32000.000000     550.000000  7950.080000     600.000000   

       rtd_price10  rtd_quantity10  rtd_price11  rtd_quantity11  \
count  1152.000000     1152.000000          0.0             0.0   
mean   5375.290000      391.725000          NaN             NaN   
std    2646.673628      242.068466          NaN             NaN   
min    2283.000000      149.900000          NaN             NaN   
25%    2984.250000      149.975000          NaN             NaN   
50%    5609.040000      385.000000          NaN             NaN   
75%    8000.080000      626.750000          NaN             NaN   
max    8000.080000      647.000000          NaN             NaN   

       rtd_rr_break_quantity1    rtd_rr_up1  rtd_rr_down1  \
count            62197.000000  62197.000000  62197.000000   
mean                55.892151     93.081316     91.973245   
std        

In [13]:
# Check if 5-minute timestamps equal to 288 for this 24-hour dataset
timestamp_counts = df['time_interval'].value_counts()
print(timestamp_counts)

# Count the number of unique timestamps
num_unique_timestamps = len(df['time_interval'].unique())
print(f"\nNumber of unique timestamps: {num_unique_timestamps}")

time_interval
24/09/2023 0:05     419
24/09/2023 0:10     419
24/09/2023 16:25    419
24/09/2023 16:20    419
24/09/2023 16:15    419
                   ... 
24/09/2023 8:00     419
24/09/2023 7:55     419
24/09/2023 7:50     419
24/09/2023 7:45     419
25/09/2023 0:00     419
Name: count, Length: 288, dtype: int64

Number of unique timestamps: 288


In [14]:
df

time_interval    delivery_date  delivery_hour    resource_id  \
0        24/09/2023 0:05  24/09/2023 0:00              1  01BTSOLEN_BAT   
1        24/09/2023 0:05  24/09/2023 0:00              1    01CNCEP_BAT   
2        24/09/2023 0:05  24/09/2023 0:00              1    01LAMAO_BAT   
3        24/09/2023 0:05  24/09/2023 0:00              1    01LIMAY_BAT   
4        24/09/2023 0:05  24/09/2023 0:00              1  01MAGAPIT_BAT   
...                  ...              ...            ...            ...   
120667  24/09/2023 23:50  24/09/2023 0:00             24   08SLWIND_G01   
120668  24/09/2023 23:55  24/09/2023 0:00             24    08PWIND_G01   
120669  24/09/2023 23:55  24/09/2023 0:00             24   08SLWIND_G01   
120670   25/09/2023 0:00  24/09/2023 0:00             24    08PWIND_G01   
120671   25/09/2023 0:00  24/09/2023 0:00             24   08SLWIND_G01   

             resource_name trading_id major_part_id_new type_id bid_type  \
0           BATAAN BATTERY       BSEI                AC     GEN      REG   
1       CONCEPCION BATTERY       UPSI               SMC     GEN      REG   
2            LAMAO BATTERY       UPSI               SMC     GEN      REG   
3            LIMAY BATTERY       UPSI               SMC     GEN      REG   
4          MAGAPIT BATTERY       UPSI               SMC     GEN      REG   
...                    ...        ...               ...     ...      ...   
120667    SAN LORENZO WIND      TAREC                AC     GEN     NSCH   
120668     PWEI NABAS WIND       PWEI             BWCUA     GEN     NSCH   
120669    SAN LORENZO WIND      TAREC                AC     GEN     NSCH   
120670     PWEI NABAS WIND       PWEI             BWCUA     GEN     NSCH   
120671    SAN LORENZO WIND      TAREC                AC     GEN     NSCH   

       nsch_type plant_type region_id zone_id  max_cap_rtd Outage_Tag  \
0                       BAT      CLUZ  NLUZON          0.5              
1                       BAT      CLUZ  NLUZON         60.0              
2                       BAT      CLUZ  NLUZON         50.0              
3                       BAT      CLUZ  NLUZON         40.0              
4                       BAT      CLUZ  NLUZON         40.0              
...          ...        ...       ...     ...          ...        ...   
120667      MUST       WIND      CVIS   PANAY         54.0              
120668      MUST       WIND      CVIS   PANAY         36.7              
120669      MUST       WIND      CVIS   PANAY         54.0              
120670      MUST       WIND      CVIS   PANAY         36.7              
120671      MUST       WIND      CVIS   PANAY         54.0              

        outage_capacity_rtd  rtd_price1  rtd_quantity1  rtd_price2  \
0                       NaN         NaN            NaN         NaN   
1                       NaN         NaN            NaN         NaN   
2                       NaN    -10000.0          -50.0    -10000.0   
3                       NaN    -10000.0          -40.0    -10000.0   
4                       NaN         NaN            NaN         NaN   
...                     ...         ...            ...         ...   
120667                  NaN         NaN            NaN         NaN   
120668                  NaN         NaN            NaN         NaN   
120669                  NaN         NaN            NaN         NaN   
120670                  NaN         NaN            NaN         NaN   
120671                  NaN         NaN            NaN         NaN   

        rtd_quantity2  rtd_price3  rtd_quantity3  rtd_price4  rtd_quantity4  \
0                 NaN         NaN            NaN         NaN            NaN   
1                 NaN         NaN            NaN         NaN            NaN   
2                 0.0     32000.0           50.0         NaN            NaN   
3                 0.0     32000.0           40.0         NaN            NaN   
4                 NaN         NaN            NaN         NaN            NaN   
...  

In [15]:
# Assuming your DataFrame is named df
resource_id_counts = df['resource_id'].value_counts()

# Set display options to show all rows and columns
pd.set_option('display.max_rows', None)

# Display the unique names and their counts
resource_id_counts

resource_id
01BTSOLEN_BAT     288
06HPCO_G01        288
11PACERM_G01      288
11NBPC_G01        288
11MNRGY_G02       288
11MNRGY_G01       288
11KEGMAR_G01      288
10MEGC_G01        288
10IDPP_G01        288
09WMPC_U10        288
09WMPC_U09        288
09WMPC_U08        288
09WMPC_U07        288
09WMPC_U06        288
09WMPC_U05        288
09WMPC_U04        288
09WMPC_U03        288
09WMPC_U02        288
09WMPC_U01        288
09PANABNK_U02     288
09PANABNK_U01     288
09KEGJIM_G01      288
14MARBEL_U01      288
14ALMADA_G01      288
13TUDAY2_G01      288
11PKBUK_G01       288
12KEGGIN_G01      288
12KEGTAN_G01      288
11KBSOL_G01       288
06CABI_G01        288
06BISCOM_G01      288
07UBAY_BAT        288
06KABAN_BAT       288
05TOLEDO_BAT      288
04ORMOC_BAT       288
14NVSOL_G01       288
14ASTROSOL_G01    288
13DIGSOL_G01      288
11KIRAS_G01       288
14SUPKOR_G01      288
12NACSUR_G01      288
14PKPSOC_G01      288
13TM1_U02         288
13TM1_U01         288
13MPCDIG_G01      28

In [16]:
# Reset display options to default
pd.set_option('display.max_rows', 10)
# pd.set_option('display.max_columns', 20)

In [17]:
num_unique_resource_ids = df['resource_id'].nunique()
num_unique_resource_ids

419


There are 419 participants, with each having an entry for a timestamp. 
419 * 288 5-min intervals in 24h = 120,672 total rows.

## Convert to date-time

In [18]:
# Convert 'time_interval' to datetime format
df['time_interval'] = pd.to_datetime(df['time_interval'], format='%d/%m/%Y %H:%M')

# Filtering

In [23]:
# Group by 'resource_id' and count the non-NaN entries in 'rtd_quantity1'
counts = df.groupby('resource_id')['rtd_quantity8'].count()

# Sort the counts in descending order and select the top 20
top_resources = counts.sort_values(ascending=False).head(20)
# Display the top 20 resource_id with non-NaN counts
for resource_id, count in top_resources.items():
    print(f"Resource ID: {resource_id}, Non-NaN Count: {count}")

Resource ID: 01SUAL_G01, Non-NaN Count: 288
Resource ID: 13SMC_U02, Non-NaN Count: 288
Resource ID: 10GNPK_U03, Non-NaN Count: 288
Resource ID: 10GNPK_U02, Non-NaN Count: 288
Resource ID: 03PAGBIL_G03, Non-NaN Count: 288
Resource ID: 01MSINLO_G01, Non-NaN Count: 288
Resource ID: 01MSINLO_G03, Non-NaN Count: 288
Resource ID: 13SMC_U01, Non-NaN Count: 288
Resource ID: 01MSINLO_G02, Non-NaN Count: 288
Resource ID: 01SUAL_G02, Non-NaN Count: 288
Resource ID: 01HYPGRN_G01, Non-NaN Count: 268
Resource ID: 11FDC_U02, Non-NaN Count: 120
Resource ID: 11FDC_U01, Non-NaN Count: 96
Resource ID: 11STEAG_U01, Non-NaN Count: 96
Resource ID: 08PDPP3_H, Non-NaN Count: 0
Resource ID: 06STNEGB_G01, Non-NaN Count: 0
Resource ID: 07BDPP_U03, Non-NaN Count: 0
Resource ID: 07BDPP_U02, Non-NaN Count: 0
Resource ID: 07BDPP_U01, Non-NaN Count: 0
Resource ID: 06VMC_G02, Non-NaN Count: 0


In [24]:
pd.set_option('display.max_columns', None)

# Filter by 'resource_id' and time range
filtered_df = df[(df['resource_id'] == '01SUAL_G01')]

# Select specific columns
selected_columns = ['time_interval', 'resource_id', 'resource_name', 'trading_id', 'major_part_id_new',
                     'plant_type', 'region_id', 'zone_id', 'max_cap_rtd',
                     'rtd_price1', 'rtd_quantity1', 'rtd_price2', 'rtd_quantity2',
                     'rtd_price3', 'rtd_quantity3', 'rtd_price4', 'rtd_quantity4',
                     'rtd_price5', 'rtd_quantity5', 'rtd_price6', 'rtd_quantity6',
                     'rtd_price7', 'rtd_quantity7', 'rtd_price8', 'rtd_quantity8',
                     'rtd_price9', 'rtd_quantity9', 'rtd_price10', 'rtd_quantity10',
                     'rtd_price11', 'rtd_quantity11', 'max_offer_energy_rtd_price',
                     'max_offer_energy_rtd', 'energy_dipc', 'energy_dipc_price',
                     'energy_rtx', 'energy_rtx_price', 'energy_dipc_final',
                     'energy_dipc_final_price', 'pricing_condition_final']

# Apply the selection to the filtered DataFrame
result_df = filtered_df[selected_columns]

# Sort by 'time_interval'
result_df = result_df.sort_values(by='time_interval')
result_df

time_interval resource_id resource_name trading_id  \
8080  2023-09-24 00:05:00  01SUAL_G01    SUAL CFTPP       SMEC   
8110  2023-09-24 00:10:00  01SUAL_G01    SUAL CFTPP       SMEC   
8140  2023-09-24 00:15:00  01SUAL_G01    SUAL CFTPP       SMEC   
8170  2023-09-24 00:20:00  01SUAL_G01    SUAL CFTPP       SMEC   
8200  2023-09-24 00:25:00  01SUAL_G01    SUAL CFTPP       SMEC   
...                   ...         ...           ...        ...   
16570 2023-09-24 23:40:00  01SUAL_G01    SUAL CFTPP       SMEC   
16600 2023-09-24 23:45:00  01SUAL_G01    SUAL CFTPP       SMEC   
16630 2023-09-24 23:50:00  01SUAL_G01    SUAL CFTPP       SMEC   
16660 2023-09-24 23:55:00  01SUAL_G01    SUAL CFTPP       SMEC   
16690 2023-09-25 00:00:00  01SUAL_G01    SUAL CFTPP       SMEC   

      major_part_id_new plant_type region_id zone_id  max_cap_rtd  rtd_price1  \
8080                SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
8110                SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
8140                SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
8170                SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
8200                SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
...                 ...        ...       ...     ...          ...         ...   
16570               SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
16600               SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
16630               SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
16660               SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   
16690               SMC       COAL      CLUZ  NLUZON        647.0    -10000.0   

       rtd_quantity1  rtd_price2  rtd_quantity2  rtd_price3  rtd_quantity3  \
8080             0.0    -10000.0          250.0     5200.08          300.0   
8110             0.0    -10000.0          250.0     5200.08          300.0   
8140             0.0    -10000.0          250.0     5200.08          300.0   
8170             0.0    -10000.0          250.0     5200.08          300.0   
8200             0.0    -10000.0          250.0     5200.08          300.0   
...              ...         ...            ...         ...            ...   
16570            0.0    -10000.0          250.0     5000.08          300.0   
16600            0.0    -10000.0          250.0     5000.08          300.0   
16630            0.0    -10000.0          250.0     5000.08          300.0   
16660            0.0    -10000.0          250.0     5000.08          300.0   
16690            0.0    -10000.0          250.0     5000.08          300.0   

       rtd_price4  rtd_quantity4  rtd_price5  rtd_quantity5  rtd_price6  \
8080      5700.08          350.0     6000.08          400.0     6500.08   
8110      5700.08          350.0     6000.08          400.0     6500.08   
8140      5700.08          350.0     6000.08          400.0     6500.08   
8170      5700.08          350.0     6000.08          400.0     6500.08   
8200      5700.08          350.0     6000.08          400.0     6500.08   
...           ...            ...         ...            ...         ...   
16570     6000.08          350.0     6100.08          400.0     6500.08   
16600     6000.08          350.0     6100.08          400.0     6500.08   
16630     6000.08          350.0     6100.08          400.0     6500.08   
16660     6000.08          350.0     6100.08          400.0     6500.08   
16690     6000.08          350.0     6100.08          400.0     6500.08   

       rtd_quantity6  rtd_price7  rtd_quantity7  rtd_price8  rtd_quantity8  \
8080           450.0     7000.08          500.0     7400.08          550.0   
8110           450.0     7000.08          500.0     7400.08          550.0   
8140           450.0     7000.08          500.0     7400.08          550.0   
8170           450.0     7000.08          500.0     7400.08          550.0   
8200           450